jsonlinter.py

```diff
@@ -0,0 +1,98 @@
+# python3 lintjson.py
+# William Rehwinkel / obiesource, 2022
+
+import sys
+import os
+import json
+import traceback
+
+def getJsonFiles():
+    # list all files recursively borrowed from https://pythonexamples.org/python-get-list-of-all-files-in-directory-and-sub-directories/
+    filelist = list()
+    path = "./"
+    
+    for root, dirs, files in os.walk(path):
+	    for file in files:
+            #append the file name to the list
+		    filelist.append(os.path.join(root,file))
+    
+    #print all the file names
+    jsonFiles = list()
+    
+    for name in filelist:
+        if name.lower().endswith(".json"):
+            jsonFiles.append(name)
+    return jsonFiles
```

Comment:
I think this function can be simplified using `glob` (found [here](https://stackoverflow.com/a/40755802)).

```suggestion
from glob import glob

def getJsonFiles():
    return glob("./**/*.json", recursive=True)
```

Reply:
While I'd run with the glob solution,
![you did a great job](https://www.meme-arsenal.com/memes/9f800883f35db2025f669efdd3fdd9bb.jpg)

A note on your solution @FiskFan1999, out of interest, for you and anyone else who comes upon this later:
A summary of your process:

- Iterate over every file in the directory and add them to the list `filelist`
- Iterate over every item in `filelist` and names that pass your filter get added to `jsonfiles`

A similar solution:

- Iterate over every file in the directory / and names that pass your filter get added to `jsonfiles`

Your solution closely matches the second solution in time efficiency. Another consideration is memory efficiency, where since your solution has an extra list, it's no more than half as memory efficient.

In this case I'd use the second solution. But, there are two circumstances I can think of that would make your solution preferred.

1. If we needed `filelist` to perform more than one purpose. For example, if we really did need to know all the files regardless of whether they were `.json`s. In particular, if our other reason for needing this could not also be incorporated into the first loop. If we needed a list of all the `.md`s, we could populate both that list and `jsonfiles` in the same loop, still without a `filelist`.
2. If for some reason we could not begin to construct `jsonfiles` without having already iterated over the entire list of files. Like, if we want to tack a file number string like "file 52 of 739" onto every filename, we need to know there are 739 files. There are better examples (because we still wouldn't need `filelist` just to count the files) but that's the idea

There's more to be said here but this comment already is making a huge assumption about how much anyone cares 😅 Hope this is thought provoking.